src/app/src/Map.js

```diff
@@ -1,19 +1,61 @@
 import React, { Component } from 'react';
-import ReactMapGL, { NavigationControl } from 'react-map-gl';
+import ReactMapGL, {
+    NavigationControl,
+    Marker,
+    FlyToInterpolator,
+} from 'react-map-gl';
 import 'mapbox-gl/dist/mapbox-gl.css';
 import { connect } from 'react-redux';
 
-import onChangeViewport from './map.actions';
+import { onChangeViewport } from './map.actions';
 import './Map.scss';
+import SensorMarker from './SensorMarker';
+import BackToMapButton from './BackToMapButton';
+import sensors from './sensors.json';
+import { toggleBackToMapButton } from './map.actions';
 
 class GLMap extends Component {
+    constructor(props) {
+        super(props);
+
+        this.goToLocation = this.goToLocation.bind(this);
+        this.renderCityMarkers = this.renderCityMarkers.bind(this);
+    }
+
+    goToLocation(options) {
+        toggleBackToMapButton();
+
+        this.props.updateViewport(
+            Object.assign(options, {
```

Comment:
Heads up that this will mutate the `options` object. Using `Object.assign({}, options, { ...` won't.

Conceivably this could also make use of immutability helper syntax.

Reply:
Good catch.